internal/btf/btf_test.go

```diff
@@ -12,6 +12,49 @@ import (
 	"github.com/cilium/ebpf/internal/testutils"
 )
 
+func TestFindType(t *testing.T) {
+	fh, err := os.Open("testdata/vmlinux-btf.gz")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer fh.Close()
+
+	rd, err := gzip.NewReader(fh)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rd.Close()
+
+	spec, err := loadRawSpec(rd, binary.LittleEndian, nil, nil)
+	if err != nil {
+		t.Fatal("Can't load BTF:", err)
+	}
+
+	// spec.FindType MUST fail if typ is not a non-nil **T, where T satisfies btf.Type.
+	for _, typ := range []interface{}{
+		nil,
+		Struct{},
+		&Struct{},
+	} {
```

Comment:
Add something like `**int`?

Reply:
Yes, and now that I think about it, we should also check something like slices and maps as "strange" types.